### PR TITLE
Adds in updateSubscriptions functionality

### PR DESCRIPTION
**Pull request by davidtruong**

### CODE REVIEW

File: Iterable-iOS-SDK/IterableAPI.m [hunk 1]
```diff
@@ -861,6 +861,25 @@ - (void)trackPurchase:(NSNumber *)total items:(NSArray<CommerceItem *> *)items d
 }
 
 // documented in IterableAPI.h
+- (void)updateSubscriptions:(NSArray *)emailListIds unsubscribedChannelIds:(NSArray *)unsubscribedChannelIds unsubscribedMessageTypeIds:(NSArray *)unsubscribedMessageTypeIds
+{
+    NSMutableDictionary *mutableDictionary = [[NSMutableDictionary alloc] init];
+    
+    if (_email != nil) {
```
Comment (dontgitit) [inline Iterable-iOS-SDK/IterableAPI.m]: we should refactor this to some `addCommonFieldsToDictionary:(NSMutableDictionary dictionary)` method

File: Iterable-iOS-SDK/IterableAPI.m [hunk 2]
```diff
@@ -937,4 +956,19 @@ -(void) showSystemNotification:(NSString *)title body:(NSString *)body buttonLef
...
+ 
+ @abstract Helper function to add a kvp to a dictionary
+ 
+ @param callbackBlock  Callback ITEActionBlock
+ 
+ */
+- (void)tryAddValueToMutableDictionary:(NSMutableDictionary *)mutableDictionary key:(NSString *)key value:(NSObject *)value
+{
+    if (mutableDictionary != nil && value != nil && key != nil) {
+        [mutableDictionary setObject:value  forKey:key];
```
Comment (dontgitit) [inline Iterable-iOS-SDK/IterableAPI.m]: extra space between value and forKey

File: Iterable-iOS-SDK/IterableConstants.h [hunk 3]
```diff
@@ -29,6 +30,8 @@ extern NSString *const ITBL_KEY_SEND_AT;
 extern NSString *const ITBL_KEY_TOKEN;
 extern NSString *const ITBL_KEY_TEMPLATE_ID;
 extern NSString *const ITBL_KEY_TOTAL;
+extern NSString *const ITBL_KEY_UNSUB_CHANNEL;
```
Comment (dontgitit) [inline Iterable-iOS-SDK/IterableConstants.h:33]: do all these keys really need to be public/in the `.h`? I feel like they're internal to our API and don't particularly need to be exposed... or is this just the way it is in obj-c? can't remember...

**Review by dontgitit — APPROVED** [review]

File: Iterable-iOS-SDK/IterableAPI.m (hunk 1 above)
Comment (davidtruong) [reply Iterable-iOS-SDK/IterableAPI.m]: Added in addEmailOrUserIdToDictionary to make it consistent with the Android helper function.

**Pull request merged**